log4j-api-test/src/test/java/org/apache/logging/log4j/message/ParameterizedMessageTest.java

```diff
@@ -185,16 +185,73 @@ void testSerializable(final Object arg) {
         assertThat(actual.getFormattedMessage()).isEqualTo(expected.getFormattedMessage());
     }
 
+    @Test
+    void formatToWithMoreArgsButNoWarn() {
+        final String pattern = "no warn {} {}";
+        final String expectedMessage = "no warn a b";
+        final Object[] args = {"a", "b", new RuntimeException()};
+        final ParameterizedMessage message = new ParameterizedMessage(pattern, args);
+        final StringBuilder buffer = new StringBuilder();
+        message.formatTo(buffer);
+        assertThat(buffer.toString()).isEqualTo(expectedMessage);
+        assertThat(message.getThrowable()).isInstanceOf(RuntimeException.class);
+        final List<StatusData> statusDataList = statusListener.getStatusData().collect(Collectors.toList());
+        assertThat(statusDataList).hasSize(0);
+    }
+
+    @Test
+    void formatWithMoreArgsButNoWarn() {
+        final String pattern = "no warn {} {}";
+        final String expectedMessage = "no warn a b";
+        final Object[] args = {"a", "b", new RuntimeException()};
+        final String message = ParameterizedMessage.format(pattern, args);
+        assertThat(message).isEqualTo(expectedMessage);
+        final List<StatusData> statusDataList = statusListener.getStatusData().collect(Collectors.toList());
+        assertThat(statusDataList).hasSize(0);
+    }
+
+    /**
+     * In this test cases, constructed the following scenarios: <br>
+     * <p>
+     * 1. The placeholders are greater than the count of arguments. <br>
+     * 2. The placeholders are less than the count of arguments. <br>
+     * 3. The arguments contains an exception, and the placeholder is greater than normal arguments. <br>
+     * 4. The arguments contains an exception, and the placeholder is less than the arguments.<br>
+     * All of these should logged in status logger with WARN level.
+     * </p>
+     *
+     * @return streams
+     */
     static Stream<Object[]> testCasesForInsufficientFormatArgs() {
-        return Stream.of(new Object[] {1, "foo {}"}, new Object[] {2, "bar {}{}"});
+        return Stream.of(
+                new Object[] {"more {} {}", 2, new Object[] {"a"}, "more a {}"},
+                new Object[] {"more {} {} {}", 3, new Object[] {"a"}, "more a {} {}"},
+                new Object[] {"less {}", 1, new Object[] {"a", "b"}, "less a"},
+                new Object[] {"less {} {}", 2, new Object[] {"a", "b", "c"}, "less a b"},
+                new Object[] {
+                    "more throwable {} {}",
+                    2,
+                    new Object[] {"a", new RuntimeException()},
+                    "more throwable a java.lang.RuntimeException"
+                },
```

Comment:
This case should issue no warning: the `RuntimeException` is simply classified as normal argument and it fills the second placeholder.

Reply:
I will adjust this case.